server/game/cards/10-TEL/ArdentOmoidasu.js

```diff
@@ -0,0 +1,21 @@
+const DrawCard = require('../../drawcard.js');
+const AbilityDsl = require('../../abilitydsl');
+
+class ArdentOmoidasu extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            title: 'Steal 2 honor',
+            when: {
+                onCardDishonored: (event, context) => event.card.controller === context.player
+                    && event.context.player === context.player.opponent
```

Comment:
I think you should check specifically what the card says (i.e. is this an opponent's card or ring effect?) rather than assuming that checking context.player will cover all eventualities.

Reply:
Akodo Motivator is a good model to follow here, I think :)